target/scripts/build/packages.sh

```diff
@@ -0,0 +1,101 @@
+#! /bin/bash
+
+set -eE -u -o pipefail
+
+# shellcheck source=../helpers/log.sh
+source /usr/local/bin/helpers/log.sh
+
+export DEBIAN_FRONTEND=noninteractive
+QUIET='-qq' ; _log_level_is_trace && QUIET='-y'
+
+function _pre_installation_steps() {
+  _log 'info' 'Starting package installation'
+  _log 'debug' 'Running pre-installation steps'
+
+  _log 'trace' 'Updating package signatures'
+  apt-get "${QUIET}" update
+
+  _log 'trace' 'Installing packages that are needed early'
+  apt-get "${QUIET}" install --no-install-recommends \
+    apt-transport-https apt-utils ca-certificates \
+    curl gnupg2 ssl-cert 2>/dev/null
+
+  _log 'trace' 'Upgrading packages'
+  apt-get "${QUIET}" dist-upgrade
+
+  _log 'trace' 'Updating CA certificates'
+  update-ca-certificates
+}
+
+function _install_postfix() {
+  _log 'debug' 'Installing Postfix'
+
+  _log 'warn' "Applying workaround for Postfix bug (see https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403)"
+
+  mv /bin/hostname /bin/hostname.bak
+  echo "echo 'docker-mailserver.invalid'" >/bin/hostname
+  chmod +x /bin/hostname
+  apt-get "${QUIET}" install --no-install-recommends postfix
+  mv /bin/hostname.bak /bin/hostname
+}
+
+function _setup_apt_sources() {
+  _log 'debug' 'Acquiring package sources (PPAs)'
+
+  _log 'trace' 'Getting RSPAMD GPG keys and PPA'
+  local RSPAMD_GPG_KEY_LOCATION='/etc/apt/trusted.gpg.d/rspamd.gpg'
+  curl -sSfL https://rspamd.com/apt-stable/gpg.key \
+    | gpg --dearmor >"${RSPAMD_GPG_KEY_LOCATION}"
+  echo \
+    "deb [arch=amd64 signed-by=${RSPAMD_GPG_KEY_LOCATION}] http://rspamd.com/apt-stable/ bullseye main" \
+    >/etc/apt/sources.list.d/rspamd.list
+
+  _log 'trace' 'Updating package signatures again'
+  apt-get "${QUIET}" update
+}
+
+function _install_packages() {
+  _log 'debug' 'Installing all packages now'
+
+  declare -a DOVECOT_PACKAGES
+  DOVECOT_PACKAGES=(
+    dovecot-core
+    dovecot-fts-xapian
+    dovecot-imapd
+    dovecot-ldap
+    dovecot-lmtpd
+    dovecot-managesieved
+    dovecot-pop3d
+    dovecot-sieve
+    dovecot-solr
+  )
+
+  apt-get "${QUIET}" --no-install-recommends install \
+    amavisd-new binutils cron dbconfig-no-thanks \
+    clamav clamav-daemon \
+    "${DOVECOT_PACKAGES[@]}" dumb-init \
+    fetchmail fail2ban gzip htop iproute2 \
+    locales logwatch libldap-common libsasl2-modules netcat-openbsd nftables \
+    opendkim opendkim-tools opendmarc \
+    pflogsumm postgrey p7zip-full \
+    postfix-ldap postfix-pcre postfix-policyd-spf-python \
+    postsrsd pyzor razor redis-server rspamd rsyslog sasl2-bin \
+    spamassassin supervisor uuid xz-utils
```

Comment:
Likewise, it would be good to separate packages by features. It's not always clear why we have some packages (_and with time gets harder to track via `git blame`_).

- I know that `uuid` was added for the current file locking scripts.
- `clamav` + `clamav-daemon` could be by itself or for the time being mixed with `amavisd-new` + `spamassassin`. I think `p7zip-full` is for `amavisd-new` support.
- `libldap-common`, `postfix-ldap`, `libsasl2-modules`, `sasl2-bin` for LDAP (_and Cyrus based SASL? One might not be_ :man_shrugging: )
- `redis-server` stood out. This DB package is for `rspamd` I assume? (_highlights why I'd like the separation / scoping to features/dependencies_)
- You've dropped quite a few packages...why? (**EDIT:** [Mentioned in this commit](https://github.com/docker-mailserver/docker-mailserver/pull/2752/commits/f9961baf2861ad53e5a421b1ad0e0621c27a79ec)) These provide support to Amavis AFAIK, it lists all the supported codecs found during the startup logs, ClamAV maybe also uses it to check for viruses? Is this something you can confidently drop? (_with proper stage separation, we can halve the weight of the image if we publish an alternative antivirus tag variant, or the other way around with a lightweight `core` base_)

Some of the packages dropped aren't related to the archive support that the commit messages justifies the removal with. **I am not comfortable this being part of the PR.**

Please maintain the existing package list as-is, and defer the `rspamd` ones to a separate PR for introducing `rspamd` support. A separate PR may follow this one to introduce the breaking change of removing packages, and that is all that PR should cover, preferably the scoping/grouping would already be laid out here.

Reply:
I see. I will not touch the packages :)

PS: You're right: `redis-server` is for `rspamd`